ec2/al2-managed.yaml

```diff
@@ -0,0 +1,846 @@
+Metadata:
+  'AWS::CloudFormation::Interface':
+    ParameterGroups:
+    - Label:
+        default: 'Parent Stacks'
+      Parameters:
+      - ParentVPCStack
+      - ParentSSHBastionStack
+      - ParentAlertStack
+      - ParentKmsKeyStack
+      - ParentZoneStack
+      - ParentClientStack1
+      - ParentClientStack2
+      - ParentClientStack3
+    - Label:
+        default: 'Network Parameters'
+      Parameters:
+      - SubnetName
+      - SubDomainNameWithDot
+      - IngressTcpPort1
+      - IngressTcpPort2
+      - IngressTcpPort3
+    - Label:
+        default: 'Permission Parameters'
+      Parameters:
+      - KeyName
+      - IAMUserSSHAccess
+      - ManagedPolicyArns
+    - Label:
+        default: 'EC2 Parameters'
+      Parameters:
+      - Name
+      - AmazonLinux2Version
+      - RestoreImageId
+      - InstanceType
+      - UserData
+      - RootVolumeSize
+    - Label:
+        default: 'Operational Parameters'
+      Parameters:
+      - LogsRetentionInDays
+      - MaintenanceWindowSchedule
+      - MaintenanceWindowScheduleTimezone
+      - BackupRetentionPeriod
+      - BackupScheduleExpression
+Parameters:
+  ParentVPCStack:
+    Description: 'Stack name of parent VPC stack based on vpc/vpc-*azs.yaml template.'
+    Type: String
+  ParentSSHBastionStack:
+    Description: 'Optional but recommended stack name of parent SSH bastion host/instance stack based on vpc/vpc-*-bastion.yaml template.'
+    Type: String
+    Default: ''
+  ParentAlertStack:
+    Description: 'Optional but recommended stack name of parent alert stack based on operations/alert.yaml template.'
+    Type: String
+    Default: ''
+  ParentKmsKeyStack:
+    Description: 'Optional stack name of parent KMS key stack based on security/kms-key.yaml template.'
+    Type: String
+    Default: ''
+  ParentZoneStack:
+    Description: 'Optional stack name of parent zone stack based on vpc/zone-*.yaml template.'
+    Type: String
+    Default: ''
+  ParentClientStack1:
+    Description: 'Optional stack name of parent Client Security Group stack based on state/client-sg.yaml template to allow network access from the EC2 instance to whatever uses the client security group.'
+    Type: String
+    Default: ''
+  ParentClientStack2:
+    Description: 'Optional stack name of parent Client Security Group stack based on state/client-sg.yaml template to allow network access from the EC2 instance to whatever uses the client security group.'
+    Type: String
+    Default: ''
+  ParentClientStack3:
+    Description: 'Optional stack name of parent Client Security Group stack based on state/client-sg.yaml template to allow network access from the EC2 instance to whatever uses the client security group.'
+    Type: String
+    Default: ''
+  KeyName:
+    Description: 'Optional key pair of the ec2-user to establish a SSH connection to the EC2 instance.'
+    Type: String
+    Default: ''
+  IAMUserSSHAccess:
```

Comment:
Should we deprecate IAM User SSH Access in favor of EC2 Instance Connect?

Reply:
the reason why I kept it that the bastion host use the same approach. But I think that most people want to use SSM Session Manager today?